### PR TITLE
Better input validation when asking for file extensions

### PR DESCRIPTION
**Pull request by jschumme**
# Justification
We need better input validation when prompting for file extensions:

image

# Implementation
Basically two RegEx expressions

# Testing
Test for validity regex algorithm added

### CODE REVIEW

File: src/commands.ts [hunk 1]
```diff
@@ -226,13 +226,22 @@ async function readOrRequestFileExtensionConfig(uri: vscode.Uri): Promise<string
...
     if (!extensions) {
-        extensions = await vscu.showInputBox(
-            'Please enter the file extensions your DataPlugin can handle in the right syntax: ',
-            '*.tdm; *.xls ...'
-        );
+        // eslint-disable-next-line no-constant-condition
+        while (true) {
+            // eslint-disable-next-line no-await-in-loop
+            extensions = await vscu.showInputBox(
+                'Please enter the file extensions your DataPlugin can handle in the right syntax: ',
```
Comment (LukasKall) [inline src/commands.ts:233]: Reading the text, it's not really clear to me that you can enter multiple extensions. And the question "What is the right syntax?" may occur. What do you think about

`Please enter one or multiple file extensions your DataPlugin can handle in the shown format: `

Or maybe @GrKuehnH has a better idea.

File: src/vscode-utils.ts [hunk 2]
```diff
@@ -58,6 +58,23 @@ export function isDocumentEmpty(): boolean {
...
 }
 
+export function isValidFileExtensionInput(input: string): boolean {
+    const regExSingleExtension = new RegExp(/^\*\.[a-zA-Z]*(;)*/gm); // *.csv || *.csv;
+    const regExMultipleExtensions = new RegExp(/^\*\.[a-zA-Z]*(;)(\s)?(\*\.[a-zA-Z]*(;)*)*/gm); // *.csv;*.txt || *.csv;*.txt; || *.csv; *.txt;
+
+    const matchSingleExtension = input?.match(regExSingleExtension);
+    const matchMultipleExtensions = input?.match(regExMultipleExtensions);
+
+    let matchArray = matchSingleExtension;
```
Comment (LukasKall) [inline src/vscode-utils.ts:68]: I would usually avoid using the type of a variable in its name. What about just `matches`?

File: src/commands.ts (hunk 1 above)
Comment (jschumme) [reply src/commands.ts:233]: The placeholder text should give a hint what the right syntax looks like. See screenshot that I attached to the PR. I'm okay with your text proposal. Maybe we find something better for "can handle". @GrKuehnH thanks!

File: src/vscode-utils.ts (hunk 2 above)
Comment (jschumme) [reply src/vscode-utils.ts:68]: yes 👍

File: src/commands.ts (hunk 1 above)
Comment (GrKuehnH) [reply src/commands.ts:233]: Suggestion: Enter the file extension(s) your DataPlugin handles using the following syntax:

I do remember, however, that we were told some years ago that we weren't supposed to use the asterisk when writing filenames in documentation: .xml; .tdm.  Does this also apply to software?

Comment (jschumme) [reply src/commands.ts:233]: I stuck with the rules that DIAdem specified for the file extension input:

image

I like your suggestion @GrKuehnH !

**Review by LukasKall — APPROVED** [review]

**Review by EvaTi — APPROVED** [review]

**Pull request merged**